File: polygon/src/main/java/com/xattacker/android/view/polygon/RegionMark.java
package com.xattacker.android.view.polygon;

import android.graphics.PointF;

import com.google.gson.annotations.SerializedName;

public final class RegionMark
{
	final static float MARK_RADIUS = 15f;

	@SerializedName("id")
	private String _id;
	
	@SerializedName("title")
	String _title;
	
	@SerializedName("position")
	PointF _position;
	
	@SerializedName("color")
	private RegionColor _color;
	
	PolygonRegion _belongRegion;
	
	public String getId()
	{
		return _id;
	}

	public void setId(String aId)
	{
		_id = aId;
	}

	public String getTitle()
	{
		return _title;
	}

	public void setTitle(String aTitle)
	{
		_title = aTitle;
	}

	public PointF getPosition()
	{
		return _position;
	}

	public void setPosition(PointF aPosition)
	{
		_position = aPosition;
	}

	public RegionColor getColor()
	{
		return _color;
	}

	public void setColor(RegionColor aColor)
	{
		_color = aColor;
	}
	
	public boolean isPointInMark(PointF aPoint)
   {
		 boolean hit = false;
		 float size = MARK_RADIUS*2;
		 PointF tl = new PointF(_position.x - size, _position.y - size);
		 PointF br = new PointF(_position.x + size, _position.y + size);
		 
		 if (
			 aPoint.x >= tl.x && aPoint.x <= br.x &&
			 aPoint.y >= tl.y && aPoint.y <= br.y
			 )
		 {
			 hit = true;
		 }
		 		 
		 return hit;
   }
}

File: polygon/src/main/java/com/xattacker/android/view/polygon/RegionMark.kt
package com.xattacker.android.view.polygon

import android.graphics.PointF

import com.google.gson.annotations.SerializedName

class RegionMark
{
    @SerializedName("id")
    var id: String? = null

    @SerializedName("title")
    var title: String? = null

    @SerializedName("position")
    var position: PointF = PointF()

    @SerializedName("color")
    var color: RegionColor? = null

    internal var _belongRegion: PolygonRegion? = null

    fun isPointInMark(point: PointF): Boolean
    {
        var hit = false
        val size = MARK_RADIUS * 2
        val tl = PointF(position.x - size, position.y - size)
        val br = PointF(position.x + size, position.y + size)

        if (point.x >= tl.x && point.x <= br.x &&
            point.y >= tl.y && point.y <= br.y)
        {
            hit = true
        }

        return hit
    }

    companion object
    {
        internal const val MARK_RADIUS = 15f
    }
}
